src/main/java/uk/gov/companieshouse/accounts/filing/controller/TransactionController.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.companieshouse.accounts.filing.controller;
+
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import uk.gov.companieshouse.accounts.filing.exceptionhandler.ResponseException;
+import uk.gov.companieshouse.accounts.filing.exceptionhandler.UriValidationException;
+import uk.gov.companieshouse.accounts.filing.model.ValidationState;
+import uk.gov.companieshouse.accounts.filing.service.file.validation.AccountsValidationStrategy;
+import uk.gov.companieshouse.api.model.accountvalidator.AccountsValidatorStatusApi;
+import uk.gov.companieshouse.logging.Logger;
+
+@Controller
+@RequestMapping("/transactions/{transactionId}/accounts-filing/{accountsFilingId}")
+public class TransactionController {
+
+    private final Logger logger;
+    private final AccountsValidationStrategy accountValidationStrategy;
+
+    @Autowired
+    public TransactionController(AccountsValidationStrategy accountValidationStrategy, Logger logger) {
+        this.accountValidationStrategy = accountValidationStrategy;
+        this.logger = logger;
+    }
+    
+    @GetMapping("/file/{fileId}/status")
+    public ResponseEntity<ValidationState> fileValidationStatus(@PathVariable final String fileId, @PathVariable final String accountsFilingId){
+        Optional<AccountsValidatorStatusApi> accountValidationResultOptional = accountValidationStrategy.validationStatusResult(fileId);
+
+        if (accountValidationResultOptional.isEmpty())
+            return ResponseEntity.notFound().build();
+
+        AccountsValidatorStatusApi accountValidationResult = accountValidationResultOptional.get();
+        accountValidationStrategy.saveFileValidationResult(accountsFilingId, accountValidationResult);
```

Comment:
I think you could do this in one line like
```suggestion
        accountValidationStrategy.saveFileValidationResult(accountsFilingId, accountValidationResultOptional.get());
```